### DISCUSSION TITLE
stabix, slip transfer calculation, ids = gB.ebsdId; ids errors

### DISCUSSION THREAD

**Discussion by Lu-Peter** [mtex-toolbox/mtex#2024] [Ask Anything] [answered]
Dear There

I hope this is the right place to find help. I am using the [stabix](https://github.com/stabix/stabix) ( [MTEX2016_Mercier_4th_exercise_final.m](https://github.com/stabix/stabix/blob/master/MTEX2016_exercises/MTEX2016_Mercier_4th_exercise_final.m)) to calculate the m' for slip transfer. The 'ids' seems wrong. ids contain 'id's that do not belong to Ti-Hex (alpha phase). Would kindly anyone can help to correct this issue? 

Thank you all very much.
SL

Below are the code I used and EBSD info.

>> clear
>> test_2
>> [grains, ebsd.grainId] = calcGrains(ebsd('indexed'));
>> ebsd(grains(grains.grainSize<10)) = [];
>> [grains, ebsd.grainId] = calcGrains(ebsd('indexed'));
>> grains = smooth(grains,5);
>> gB = grains.boundary('Ti-Hex','Ti-Hex');
>> ids = gB.ebsdId;
>> hkil_1 = [0,1,-1,0];
uvtw_1 = [2,-1,-1,0];
hkil_2 = [0,1,-1,0];
uvtw_2 = [2,-1,-1,0];
>> nA = Miller(hkil_1(1),hkil_1(2),hkil_1(3),hkil_1(4),CS{2},'hkil');
dA = Miller(uvtw_1(1),uvtw_1(2),uvtw_1(3),uvtw_1(4),CS{2},'UVTW');
nB = Miller(hkil_2(1),hkil_2(2),hkil_2(3),hkil_2(4),CS{2},'hkil');
dB = Miller(uvtw_2(1),uvtw_2(2),uvtw_2(3),uvtw_2(4),CS{2},'UVTW');
>> nA = nA./norm(nA);
dA = dA./norm(dA);
nB = nB./norm(nB);
dB = dB./norm(dB);
>> nA_rot = rotate(nA, ebsd('Ti-Hex').orientations);
dA_rot = rotate(dA, ebsd('Ti-Hex').orientations);
nB_rot = rotate(nB, ebsd('Ti-Hex').orientations);
dB_rot = rotate(dB, ebsd('Ti-Hex').orientations);
>> mpVect = abs(dot(nA_rot(ids(:,1)), nB_rot(ids(:,2))) .* ...
         dot(dA_rot(ids(:,1)), dB_rot(ids(:,2))));
Index exceeds the number of array elements. Index must not exceed 159664.

Error in indexing (line 7)
    v.x = subsref(v.x,s(1));
 

 
>> max(ebsd.id)

ans =

      160024

 
>> ebsd
 
ebsd = EBSD
 
 Phase   Orientations     Mineral  Color  Symmetry  Crystal reference frame
     0   121 (0.076%)  notIndexed                                          
     1  159664 (100%)      Ti-Hex   Lime     6/mmm       X||a*, Y||b, Z||c*
 
 Properties: bands, bc, bs, error, mad, x, y, grainId
 Scan unit : um
 
>> size(ebsd)

ans =

      159785           1

>> length(ebsd)

ans =

      159785

>>

**Comment by kilir** [comment, marked as answer]
Hi
`ebsdId` is not the same as the index. A simpel way to overcome this is to use somethign like
```
gb = grains.boundary('ti','ti')
oris = ebsd('id',gb.ebsdId).orientations
```
and use those orienations to rotate the slip systems used as input for `mPrime()`. 

Hope that helps,

Rüdiger

**Reply by Lu-Peter** [reply]
Hi Rüdiger

I tried it. It worked. Thanks so much for your help.

Cheers
SL.
